src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -38,6 +37,32 @@ public static AnalyzeResult<RemoteResolveResult> Analyze(this GraphNode<RemoteRe
             List<DowngradeResult<RemoteResolveResult>> downgrades,
             List<GraphNode<RemoteResolveResult>> cycles)
         {
+            var workingDowngrades = RentDowngradesDictionary();
```

Comment:
Will static be null so just allocate new one; probably insignificant compared to throwing the error?